src/mqtt_client.rs

```diff
@@ -132,9 +142,10 @@ where
         Ok(read)
     }
 
-    fn write(&self, buf: &[u8]) -> Result<(), Error<N::Error>> {
+    fn write(&self, state: &mut SessionState<C>, buf: &[u8]) -> Result<(), Error<N::Error>> {
         let mut socket_ref = self.socket.borrow_mut();
         let mut socket = socket_ref.take().unwrap();
+        state.last_write_time.replace(self.clock.try_now().unwrap());
```

Comment:
If this is the only reason we need `state`, we should move `last_write_time` out of `state`.

Reply:
Why? It's a piece of state that needs to be reset when the broker connection is reset, so it seems to belong into SessionState.